plugins/jobs/src/js/JobStopRun.js

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+import gql from "graphql-tag";
+
+import JobStopRunModal from "./components/JobStopRunModal";
+
+import defaultSchema from "./data/JobModel";
+
+const stopJobRun = gql`
+  mutation {
+    stopJobRun(id: $jobId, jobRunId: $jobRunId) {
+      jobId
+    }
+  }
+`;
+
+function executeStop({ jobId, jobRunId, onSuccess }) {
+  return graphqlObservable(stopJobRun, defaultSchema, {
+    jobId,
+    jobRunId
+  })
+    .map(_ => ({ done: true }))
+    .do(_ => onSuccess())
+    .startWith({ done: false });
+}
+
+function stopOperation() {
+  const stopSubject$ = new Subject();
+  const stop$ = stopSubject$.switchMap(executeStop).catch(error => {
+    return stop$.startWith({
+      errorMsg: error.response.message,
+      done: true
+    });
+  });
+
+  return {
+    stop$,
+    stopHandler: (jobId, jobRunId, onSuccess) => {
+      stopSubject$.next({ jobId, jobRunId, onSuccess });
+    }
+  };
+}
+
+const JobStopRun = componentFromStream(props$ => {
+  const { stop$, stopHandler } = stopOperation();
+  const stopEmit$ = stop$.startWith({ done: null });
+
+  return props$
+    .combineLatest(stopEmit$, (props, stopOperation) => {
+      return { ...props, ...stopOperation };
+    })
+    .map(({ jobID, jobRuns, onClose, onSuccess, open, done }) => {
+      const disabled = done === false;
+
+      const onSuccessEvent = () => {
+        if (jobRuns.length === 1) {
```

Comment:
This is really confusing to me, why do we check for `jobRuns`?

Reply:
This is a legacy implementation, I will restore its comment.

You have a list where you can select many jobs, but the list only allows you to select one (because the implementation is not complete), so this bit test that your expectations are met, and you can only stop one job at a time. (although the nice Ux would be allowing it to do many).